src/components/ui/ProgressBarStep.js

```diff
@@ -0,0 +1,75 @@
+import { Check } from '@styled-icons/boxicons-regular'
+import React from 'react'
+import styled from 'styled-components/macro'
+
+import CircleIcon from '../ui/CircleIcon'
+import { theme } from './GlobalStyle'
+
+const PrimaryText = styled.div`
```

Comment:
Can you check if you need this styled component at all? You might be able to just move these styles into StyledCircleIcon and get rid of the wrapping div around the step number text. If that doesn't work, you can just style using the selector div {}. Creating a new one seems a bit too much.

Then you can just switch color based off status which is cleaner

Reply:
You can also access the theme inside StyledCircleIcon

Reply:
Or even style all text related stuff in StyledProgressBarStep. Then you might only have to do things like font-size once.